geniza/common/metadata_export.py

```diff
@@ -78,23 +80,51 @@ def iter_export_data_as_dicts(self):
         # save
         yield from (self.get_export_data_dict(obj) for obj in iterr)
 
-    def serialize_value(self, value):
-        """A quick serialize methodpy to transform a value into a CSV-friendly string.
+    def serialize_value(self, value: object) -> str:
+        """A quick serialize method to transform a value into a CSV-friendly string.
 
         :param value: Any value
         :type value: object
+
         :return: Stringified value
         :rtype: str
         """
         if type(value) is bool:
             return "y" if value else "n"
+        elif value is None:
+            return ""
+        elif type(value) in {list, tuple, set}:
+            return self.sep_within_cells.join(
+                self.serialize_value(subval) for subval in sorted(list(value))
+            )
         else:
             return str(value)
 
-    def serialize_dict(self, data_dict):
-        return {str(k): self.serialize_value(v) for k, v in data_dict.items()}
+    def serialize_key(self, key: object) -> str:
+        """Serialize the keys in the serialized dictionary. For now, this is identical to serializing values. Keys ought to be strings but this method enforces it.
+
+        :param key: Any object but it ought to be a string
+        :type key: object
+
+        :return: That object as a safe string
+        :rtype: str
+        """
+        return self.serialize_value(key)
+
+    def serialize_dict(self, data: dict) -> dict:
+        """Return a new dictionary whose keys and values are safe, serialized string versions of the keys and values in input dictionary `data`.
+
+        :param data: Dictionary of keys and values
+        :type data: dict
 
-    def iter_export_data_as_csv(self, fn=None, pseudo_buffer=False):
+        :return: Dictionary with keys and values safely serialized as strings
+        :rtype: dict
+        """
+        return {self.serialize_key(k): self.serialize_value(v) for k, v in data.items()}
```

Comment:
Why is `serialize_key` needed? would really rather not have the extra method call just to enforce it's a string, when we control the keys in this code. At 30k row that will add up.